app/src/common/shared/com/igalia/wolvic/search/SearchEngineWrapper.java

```diff
@@ -66,13 +56,17 @@ SearchEngineWrapper get(final @NonNull Context aContext) {
     private SearchSuggestionClient mSuggestionsClient;
     private SharedPreferences mPrefs;
     private boolean mAutocompleteEnabled;
+    private SearchEngineManager mSearchEngineManager;
+    private HashMap<String, SearchEngine> mSearchEnginesMap;
 
     private SearchEngineWrapper(@NonNull Context aContext) {
-        mContext = aContext;
+        mContext = aContext.getApplicationContext();
```

Comment:
Hmm why do we need this?

Reply:
I got a warning because holding a reference to a Context object in a static field can be a memory leak. This happens because the most common kind of Context object is an Activity, and holding a reference to it prevents it from being disposed after it has finished.

So in this case I am keeping a reference to the Context object for the whole application, which is just as useful for our needs and will have a longer lifetime.

Reply:
I have removed this change, because later in the same file we will try to cast this context directly into an activity again. This needs a better solution.